pandas/tests/groupby/test_nth.py

```diff
@@ -530,3 +530,20 @@ def test_nth_nan_in_grouper(dropna):
     )
 
     tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize("method", ["first", "last"])
+def test_first_last_with_na_object(method):
+    # https://github.com/pandas-dev/pandas/issues/32123
+    groups = pd.DataFrame({"a": [1, 1, 2, 2], "b": [1, 2, 3, pd.NA]}).groupby("a")
+    result = getattr(groups, method)()
+
+    if method == "first":
+        values = {"b": [1, 3]}
+    else:
+        values = {"b": [2, 3]}
```

Comment:
This raises a question (that perhaps should be handled in another issue).
`GroupBy.last()` returns `[2, 3]` however `GroupBy.tail(1)` returns `[2, pd.NA]`.
Is this intended behaviour? it is consistent with pandas 0.25, but undocumented.

Reply:
It seems that `last` gives the last non-null value but that doesn't seem to be well-documented like you say (probably does warrant a separate issue)